### PR TITLE
docs: Added guide for the error which occurs during installation of Kubearmor

### PR DESCRIPTION
**Pull request by VedRatan**
**Purpose of PR?**:
This  PR solves the issue of the non detection of the LSM  inside the Kubernetes running in dockerized environment
Fixes #

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: getting-started/FAQ.md [hunk 1]
```diff
@@ -289,6 +289,8 @@ EOF
 docker exec -it kind-control-plane bash -c "apt update && apt install apparmor-utils -y && systemctl restart containerd"
 ```
 
+The above command will execute the bash script to install the apparmor in the kind-control-plane, we can use the similar command to install apparmor in minikube as well and all the other docker based kubernetes tools.
```
Comment (anurag-rajawat) [inline getting-started/FAQ.md]:
```suggestion
The above command will install the AppArmor utilities in the kind-control-plane, we can also use this command to install these in minikube as well as in all the other docker based Kubernetes environments.
```

File: getting-started/deployment_guide.md [hunk 2]
```diff
@@ -284,4 +289,31 @@ kubectl exec -it $POD -- bash -c "chroot"
 ```
 Any binary other than `bash` and `nginx` would be permission denied.
 
+If you don't see Permission denied please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
+
+</details>
+
+<details>
+<summary><h4>Debug Kubearmor installation issue in dockerized Kubernetes environment</h4></summary>
+In certain scenarios, the expected behavior of the Kubernetes security tool Kubearmor might not be observed. One way to investigate this is by using the Kubearmor Command Line Interface (CLI) utility, commonly referred to as [karmor cli](https://github.com/kubearmor/kubearmor-client).
```
Comment (anurag-rajawat) [inline getting-started/deployment_guide.md]:
```suggestion
In certain scenarios, the expected behavior of KubeArmor might not be observed. One way to investigate this is by using the KubeArmor Command Line Interface (CLI) utility, commonly referred to as [karmor cli](https://github.com/kubearmor/kubearmor-client). 
```

File: getting-started/deployment_guide.md [hunk 3]
```diff
@@ -284,4 +289,31 @@ kubectl exec -it $POD -- bash -c "chroot"
...
 
+If you don't see Permission denied please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
+
+</details>
+
+<details>
+<summary><h4>Debug Kubearmor installation issue in dockerized Kubernetes environment</h4></summary>
+In certain scenarios, the expected behavior of the Kubernetes security tool Kubearmor might not be observed. One way to investigate this is by using the Kubearmor Command Line Interface (CLI) utility, commonly referred to as [karmor cli](https://github.com/kubearmor/kubearmor-client). 
+
+To check the status and configuration of Kubearmor, you can use the following command:
```
Comment (anurag-rajawat) [inline getting-started/deployment_guide.md]:
```suggestion
To check the status and configuration of KubeArmor, you can use the following command:
```

File: getting-started/deployment_guide.md [hunk 4]
```diff
@@ -284,4 +289,31 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+Kubernetes tools that run on Docker-based environments, such as Minikube (with Docker), KinD (Kubernetes in Docker), k3d, and others, may have AppArmor available but not loaded. AppArmor is one of the Linux Security Modules that Kubearmor can leverage to enforce security policies.
+
+To address this issue, Kubearmor provides a solution involving the use of bpf-lsm. BPF (Berkeley Packet Filter) is a technology that allows efficient packet filtering in the Linux kernel. Enabling support for BPF LSM ensures that Kubearmor can apply and enforce policies as expected in Dockerized environments associated with Kubernetes.
+
+So we need to enable [bpf-lsm](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#checking-and-enabling-support-for-bpf-lsm) for Kubearmor to apply and enforce policies as expected.
+
+We can also enable apparmor if we want, please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#using-kubearmor-with-kind-clusters) to follow along and enable apparmor in-case you want apparmor as a security module to enforce Kubearmor policies.
```
Comment (anurag-rajawat) [inline getting-started/deployment_guide.md]:
```suggestion
You can also enable AppArmor if you want to use it as a security module to enforce KubeArmor policies, please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#using-kubearmor-with-kind-clusters).
```

File: getting-started/deployment_guide.md [hunk 5]
```diff
@@ -284,4 +289,31 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+However, there are situations where ActiveLSM might not be assigned any value. This situation indicates that Kubearmor is unable to identify the appropriate LSM in a Dockerized environment, which is commonly used in Kubernetes setups.
+
+Kubernetes tools that run on Docker-based environments, such as Minikube (with Docker), KinD (Kubernetes in Docker), k3d, and others, may have AppArmor available but not loaded. AppArmor is one of the Linux Security Modules that Kubearmor can leverage to enforce security policies.
+
+To address this issue, Kubearmor provides a solution involving the use of bpf-lsm. BPF (Berkeley Packet Filter) is a technology that allows efficient packet filtering in the Linux kernel. Enabling support for BPF LSM ensures that Kubearmor can apply and enforce policies as expected in Dockerized environments associated with Kubernetes.
```
Comment (anurag-rajawat) [inline getting-started/deployment_guide.md]:
```suggestion
To address this issue, KubeArmor provides a solution involving the use of BPF LSM. BPF (Berkeley Packet Filter) is a technology that allows efficient packet filtering in the Linux kernel. Enabling support for BPF LSM ensures that KubeArmor can apply and enforce policies as expected in Dockerized environments associated with Kubernetes.
```

File: getting-started/deployment_guide.md [hunk 6]
```diff
@@ -284,4 +289,31 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+karmor probe
+```
+
+<img src="../.gitbook/assets/kubearmor_install_issue.png" width="768" class="center" alt="Karmor logs">
+
+When running this command, you should inspect the output for the value of **ActiveLSM** it is not assigned any value means there is no active LSM that Kubearmor could use to enforce policies and **Container Security** set to false. In normal circumstances, this value should be assigned a specific Linux Security Module (LSM) that Kubearmor uses to enforce security policies.
+
+However, there are situations where ActiveLSM might not be assigned any value. This situation indicates that Kubearmor is unable to identify the appropriate LSM in a Dockerized environment, which is commonly used in Kubernetes setups.
+
+Kubernetes tools that run on Docker-based environments, such as Minikube (with Docker), KinD (Kubernetes in Docker), k3d, and others, may have AppArmor available but not loaded. AppArmor is one of the Linux Security Modules that Kubearmor can leverage to enforce security policies.
```
Comment (anurag-rajawat) [inline getting-started/deployment_guide.md]:
```suggestion
Kubernetes tools that run on Docker-based environments, such as Minikube (with Docker), KinD (Kubernetes in Docker), k3d, and others, may have AppArmor available but not loaded. AppArmor is one of the Linux Security Modules that KubeArmor can leverage to enforce security policies.
```

File: getting-started/deployment_guide.md [hunk 7]
```diff
@@ -284,4 +289,31 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+To check the status and configuration of Kubearmor, you can use the following command:
+
+```
+karmor probe
+```
+
+<img src="../.gitbook/assets/kubearmor_install_issue.png" width="768" class="center" alt="Karmor logs">
+
+When running this command, you should inspect the output for the value of **ActiveLSM** it is not assigned any value means there is no active LSM that Kubearmor could use to enforce policies and **Container Security** set to false. In normal circumstances, this value should be assigned a specific Linux Security Module (LSM) that Kubearmor uses to enforce security policies.
```
Comment (anurag-rajawat) [inline getting-started/deployment_guide.md]:
```suggestion
When executing this command, check the output for the value of **ActiveLSM** field, if it is not assigned any value, it means that no active LSM is available for KubeArmor to enforce policies. Under normal circumstances, this value should be assigned a specific Linux Security Module (LSM) that KubeArmor uses to enforce security policies. Additionally, ensure that the **Container Security** field is set to true.
```

File: getting-started/deployment_guide.md [hunk 8]
```diff
@@ -284,4 +289,31 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+However, there are situations where ActiveLSM might not be assigned any value. This situation indicates that Kubearmor is unable to identify the appropriate LSM in a Dockerized environment, which is commonly used in Kubernetes setups.
+
+Kubernetes tools that run on Docker-based environments, such as Minikube (with Docker), KinD (Kubernetes in Docker), k3d, and others, may have AppArmor available but not loaded. AppArmor is one of the Linux Security Modules that Kubearmor can leverage to enforce security policies.
+
+To address this issue, Kubearmor provides a solution involving the use of bpf-lsm. BPF (Berkeley Packet Filter) is a technology that allows efficient packet filtering in the Linux kernel. Enabling support for BPF LSM ensures that Kubearmor can apply and enforce policies as expected in Dockerized environments associated with Kubernetes.
+
+So we need to enable [bpf-lsm](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#checking-and-enabling-support-for-bpf-lsm) for Kubearmor to apply and enforce policies as expected.
```
Comment (anurag-rajawat) [inline getting-started/deployment_guide.md]:
```suggestion
So we need to enable [bpf-lsm](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#checking-and-enabling-support-for-bpf-lsm) for KubeArmor to apply and enforce policies as expected.
```

File: getting-started/deployment_guide.md [hunk 9]
```diff
@@ -284,4 +289,31 @@ kubectl exec -it $POD -- bash -c "chroot"
 ```
 Any binary other than `bash` and `nginx` would be permission denied.
 
+If you don't see Permission denied please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
+
+</details>
+
+<details>
+<summary><h4>Debug Kubearmor installation issue in dockerized Kubernetes environment</h4></summary>
```
Comment (anurag-rajawat) [inline getting-started/deployment_guide.md]:
```suggestion
<summary><h4>Debug KubeArmor installation issue in dockerized Kubernetes environment</h4></summary>
```

**Review by anurag-rajawat — CHANGES_REQUESTED** [review]
Minor suggestions, rest everything looks fine.

File: getting-started/deployment_guide.md (hunk 8 above)
Comment (anurag-rajawat) [reply getting-started/deployment_guide.md]: Hi Ved, you missed this one.

**Comment by VedRatan** [comment]
any more suggestions or are we good to merge this PR?

File: getting-started/deployment_guide.md [hunk 10]
```diff
@@ -284,4 +289,30 @@ kubectl exec -it $POD -- bash -c "chroot"
 ```
 Any binary other than `bash` and `nginx` would be permission denied.
 
+If you don't see Permission denied please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
+
+</details>
+
+<details>
+<summary><h4>Debug KubeArmor installation issue in dockerized Kubernetes environment</h4></summary>
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]:
```suggestion
<summary><h4>Debug KubeArmor installation issue</h4></summary>
```

File: getting-started/deployment_guide.md [hunk 11]
```diff
@@ -284,4 +289,30 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+```
+karmor probe
+```
+
+<img src="../.gitbook/assets/kubearmor_install_issue.png" width="768" class="center" alt="Karmor logs">
+
+When executing this command, check the output for the value of **ActiveLSM** field, if it is not assigned any value, it means that no active LSM is available for KubeArmor to enforce policies. Under normal circumstances, this value should be assigned a specific Linux Security Module (LSM) that KubeArmor uses to enforce security policies. Additionally, ensure that the **Container Security** field is set to true.
+
+However, there are situations where ActiveLSM might not be assigned any value. This situation indicates that Kubearmor is unable to identify the appropriate LSM in a Dockerized environment, which is commonly used in Kubernetes setups.
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]:
```suggestion
However, there are situations where ActiveLSM might not be assigned any value. This situation indicates that Kubearmor is unable to identify the appropriate LSM in a environment, which is commonly used in Kubernetes setups.
```

File: getting-started/deployment_guide.md [hunk 12]
```diff
@@ -284,4 +289,30 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+A
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]:
```suggestion
```

File: getting-started/deployment_guide.md [hunk 13]
```diff
@@ -284,4 +289,30 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+To address this issue, KubeArmor provides a solution involving the use of BPF-LSM. BPF (Berkeley Packet Filter) is a technology that allows efficient packet filtering in the Linux kernel. Enabling support for BPF LSM ensures that KubeArmor can apply and enforce policies as expected in Dockerized environments associated with Kubernetes.
+
+So we need to enable [bpf-lsm](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#checking-and-enabling-support-for-bpf-lsm) for Kubearmor to apply and enforce policies as expected.
+
+You can also enable AppArmor if you want to use it as a security module to enforce KubeArmor policies, please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#using-kubearmor-with-kind-clusters).
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]: We need to explain here, that BPFLSM is only available on kernel versions above 5.8 or on RHEL >8.5 like distros.

Also just suggesting enabling AppArmor is not enough, we need to explain to them, how to check whether apparmor is available or not. Follow up with steps.

The steps explained in the kind clusters environment is only liable for dockerised env, but it is possible that this be the scenario in general k8s nodes as well.

We have seen this behaviour with CBL Mariner on Azure, that there's been a need to install apparmor utils into the node images.

We should include the updater script here which automatically detects and installs BPFLSM/AppArmor whichever is needed.

**Review by daemon1024 — CHANGES_REQUESTED** [review]
@nyrahul Can you share the updater daemonset script here?

**Comment by nyrahul** [comment]
PFA the updater script that enables bpf-lsm in the kubernetes worker nodes.

```yaml
apiVersion: v1
items:
- apiVersion: apps/v1
  kind: DaemonSet
  metadata:
    labels:
      kubearmor-app: updater
    name: updater
    namespace: kubearmor
  spec:
    revisionHistoryLimit: 10
    selector:
      matchLabels:
        kubearmor-app: updater
    template:
      metadata:
        labels:
          kubearmor-app: updater
      spec:
        containers:
        - args:
          - |
            grep "bpf" /rootfs/sys/kernel/security/lsm >/dev/null
            [[ $? -eq 0 ]] && echo "sysfs already has BPF enabled" && sleep infinity
            grep "GRUB_CMDLINE_LINUX.*bpf" /rootfs/etc/default/grub >/dev/null
            [[ $? -eq 0 ]] && echo "grub already has BPF enabled" && sleep infinity
            cat <<EOF >/rootfs/updater.sh
            #!/bin/bash
            lsmlist=\$(cat /sys/kernel/security/lsm)
            echo "current lsmlist=\$lsmlist"
            sed -i "s/^GRUB_CMDLINE_LINUX=.*$/GRUB_CMDLINE_LINUX=\"lsm=\$lsmlist,bpf\"/g" /etc/default/grub
            command -v grub2-mkconfig >/dev/null 2>&1 && grub2-mkconfig -o /boot/grub2.cfg
            command -v grub-mkconfig >/dev/null 2>&1 && grub-mkconfig -o /boot/grub.cfg
            command -v aa-status >/dev/null 2>&1 || yum install apparmor-utils -y
            command -v update-grub >/dev/null 2>&1 && update-grub
            command -v update-grub2 >/dev/null 2>&1 && update-grub2
            reboot
            EOF
            cat /rootfs/updater.sh
            chmod +x /rootfs/updater.sh
            chroot /rootfs/ /bin/bash /updater.sh
          image: debian
          command:
            - "bash"
            - "-c"
          imagePullPolicy: Always
          name: updater
          resources: {}
          securityContext:
            privileged: true
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
          volumeMounts:
          - mountPath: /rootfs
            mountPropagation: HostToContainer
            name: rootfs
            readOnly: false
        dnsPolicy: ClusterFirstWithHostNet
        hostNetwork: true
        hostPID: true
        nodeSelector:
          kubernetes.io/os: linux
        restartPolicy: Always
        schedulerName: default-scheduler
        securityContext: {}
        terminationGracePeriodSeconds: 30
        tolerations:
        - operator: Exists
        volumes:
        - hostPath:
            path: /
            type: DirectoryOrCreate
          name: rootfs
    updateStrategy:
      rollingUpdate:
        maxSurge: 0
        maxUnavailable: 1
      type: RollingUpdate
  status:
    currentNumberScheduled: 4
    desiredNumberScheduled: 4
    numberAvailable: 4
    numberMisscheduled: 0
    numberReady: 4
    observedGeneration: 1
    updatedNumberScheduled: 4
kind: List
metadata:
  resourceVersion: ""
```

File: getting-started/deployment_guide.md [hunk 14]
```diff
@@ -307,12 +307,106 @@ karmor probe
...
 
 So we need to enable [bpf-lsm](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#checking-and-enabling-support-for-bpf-lsm) for Kubearmor to apply and enforce policies as expected.
 
-You can also enable AppArmor if you want to use it as a security module to enforce KubeArmor policies, please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#using-kubearmor-with-kind-clusters).
+You can also enable AppArmor if you want to use it as a security module to enforce KubeArmor policies, please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#using-kubearmor-with-kind-clusters). There is a chance that neither AppArmor nor BPF-LSM is enabled on some nodes. 
+
+**We can include the following updater script which automatically detects and installs BPFLSM/AppArmor whichever is needed in kubernetes worker nodes.**
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]:
```suggestion
**We can apply the following manifest which automatically detects and installs BPFLSM/AppArmor whichever is needed in kubernetes worker nodes.**
```

File: getting-started/deployment_guide.md [hunk 15]
```diff
@@ -284,4 +289,124 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+```
+apiVersion: v1
+items:
+- apiVersion: apps/v1
+  kind: DaemonSet
+  metadata:
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]: Please add it as a file.

Please clean up the resource, it's a Daemonset List, it should be a normal daemonset.

File: getting-started/deployment_guide.md [hunk 16]
```diff
@@ -284,4 +289,124 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+    currentNumberScheduled: 4
+    desiredNumberScheduled: 4
+    numberAvailable: 4
+    numberMisscheduled: 0
+    numberReady: 4
+    observedGeneration: 1
+    updatedNumberScheduled: 4
+kind: List
+metadata:
+  resourceVersion: ""
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]:
```suggestion
```

**Review by daemon1024 — CHANGES_REQUESTED** [review]

**Comment by daemon1024** [comment]
Please fix DCO as well.

File: getting-started/deployment_guide.md [hunk 17]
```diff
@@ -284,4 +289,111 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+        - hostPath:
+            path: /
+            type: DirectoryOrCreate
+          name: rootfs
+    updateStrategy:
+      rollingUpdate:
+        maxSurge: 0
+        maxUnavailable: 1
+      type: RollingUpdate
+
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]:
```suggestion
```

File: getting-started/deployment_guide.md [hunk 18]
```diff
@@ -284,4 +289,111 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+You can also enable AppArmor if you want to use it as a security module to enforce KubeArmor policies, please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#using-kubearmor-with-kind-clusters). There is a chance that neither AppArmor nor BPF-LSM is enabled on some nodes. 
+
+**We can include the following updater script which automatically detects and installs BPFLSM/AppArmor whichever is needed in kubernetes worker nodes.**
+
+```
+  apiVersion: apps/v1
+  kind: DaemonSet
+  metadata:
+    labels:
+      kubearmor-app: updater
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]: Please keep it in a different file, and add a different command here
```
kubectl apply -f raw.github.updaterscript
```

File: getting-started/FAQ.md [hunk 19]
```diff
@@ -289,6 +289,8 @@ EOF
 docker exec -it kind-control-plane bash -c "apt update && apt install apparmor-utils -y && systemctl restart containerd"
```
Comment (daemon1024) [inline getting-started/FAQ.md:289]: Please update this FAQ to generic way as well. There are cases where AppArmor is availble but not enabled. We need to instruct them to install apparmor utils.

Example CBL Mariner. You can see the updater script we install apparmor utils there as well.

So the goal is to update the FAQ in a generic manner such that people can understand why do they need to enable AppArmor utils, how to check whether apparmor availability.

Why is the additional logic is needed for dockerized nodes.

File: getting-started/deployment_guide.md [hunk 20]
```diff
@@ -284,4 +289,111 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+To address this issue, KubeArmor provides a solution involving the use of BPF-LSM. BPF (Berkeley Packet Filter) is a technology that allows efficient packet filtering in the Linux kernel. Enabling support for BPF LSM ensures that KubeArmor can apply and enforce policies as expected in Dockerized environments associated with Kubernetes. Please note that BPFLSM is only available on kernel versions above 5.8 or on RHEL distros > 8.5.
+
+So we need to enable [bpf-lsm](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#checking-and-enabling-support-for-bpf-lsm) for Kubearmor to apply and enforce policies as expected.
+
+You can also enable AppArmor if you want to use it as a security module to enforce KubeArmor policies, please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#using-kubearmor-with-kind-clusters). There is a chance that neither AppArmor nor BPF-LSM is enabled on some nodes. 
+
+**We can include the following updater script which automatically detects and installs BPFLSM/AppArmor whichever is needed in kubernetes worker nodes.**
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]: Add a warning after the script that it will restart nodes.

**Review by daemon1024 — CHANGES_REQUESTED** [review]

File: getting-started/deployment_guide.md (hunk 18 above)
Comment (VedRatan) [reply getting-started/deployment_guide.md:325]: Do we need to keep this script in the Kubearmor project by creating a file in manifests folder?

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: .gitbook/assets/kubearmor_install_issue.png
Comment (nyrahul) [inline .gitbook/assets/kubearmor_install_issue.png:1]: It is best to not use image for such purpose. Can you post the output in text format?

File: getting-started/deployment_guide.md [hunk 21]
```diff
@@ -74,6 +74,8 @@ sh: 1: apt: Permission denied
 command terminated with exit code 126
 ```
 
+If you don't see Permission denied please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]: please use relative links
```suggestion
If you don't see Permission denied please refer [here](deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
```

File: getting-started/deployment_guide.md [hunk 22]
```diff
@@ -284,4 +289,37 @@ kubectl exec -it $POD -- bash -c "chroot"
 ```
 Any binary other than `bash` and `nginx` would be permission denied.
 
+If you don't see Permission denied please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
+
+</details>
+
+<details>
+<summary><h4>Debug KubeArmor installation issue</h4></summary>
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]: I do not think we should have this section in the getting-started guide. Lets discuss this in our regular huddle call.

**Review by nyrahul — CHANGES_REQUESTED** [review]
Please use relative links across everything in the PR. Please check comments for an example.

File: getting-started/deployment_guide.md [hunk 23]
```diff
@@ -178,6 +180,9 @@ cat: /run/secrets/kubernetes.io/serviceaccount/token: Permission denied
 }
 ```
 
+If you don't see Permission denied please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue.
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]: please use relative paths

File: getting-started/deployment_guide.md [hunk 24]
```diff
@@ -284,4 +289,37 @@ kubectl exec -it $POD -- bash -c "chroot"
 ```
 Any binary other than `bash` and `nginx` would be permission denied.
 
+If you don't see Permission denied please refer [here](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]: use relative paths.

File: getting-started/deployment_guide.md [hunk 25]
```diff
@@ -284,4 +289,37 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+However, there are situations where ActiveLSM might not be assigned any value. This situation indicates that Kubearmor is unable to identify the appropriate LSM in a environment, which is commonly used in Kubernetes setups.
+
+To address this issue, KubeArmor provides a solution involving the use of BPF-LSM. BPF (Berkeley Packet Filter) is a technology that allows efficient packet filtering in the Linux kernel. Enabling support for BPF LSM ensures that KubeArmor can apply and enforce policies as expected in Dockerized environments associated with Kubernetes. Please note that BPFLSM is only available on kernel versions above 5.8 or on RHEL distros > 8.5.
+
+So we need to enable [bpf-lsm](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/FAQ.md#checking-and-enabling-support-for-bpf-lsm) for Kubearmor to apply and enforce policies as expected.
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]: relative paths

**Review by nyrahul — DISMISSED** [review]
Use relative paths across all links.

File: getting-started/FAQ.md [hunk 26]
```diff
@@ -289,8 +289,36 @@ EOF
...
+```
+
+Following flags need to exist and set to `y`
+```ini
+CONFIG_SECURITY_APPARMOR=y
+CONFIG_SECURITY_APPARMOR_INTROSPECT_POLICY=y
+CONFIG_SECURITY_APPARMOR_HASH=y
+CONFIG_SECURITY_APPARMOR_HASH_DEFAULT=y
+CONFIG_SECURITY_APPARMOR_EXPORT_BINARY=y
+CONFIG_SECURITY_APPARMOR_PARANOID_LOAD=y
```
Comment (daemon1024) [inline getting-started/FAQ.md]:
```suggestion
CONFIG_SECURITY_APPARMOR=y
```

We only need this flag.

File: getting-started/FAQ.md [hunk 27]
```diff
@@ -289,8 +289,36 @@ EOF
...
+```
+
+Run the command to install apparmor:
+
+```
+apt update && apt install apparmor-utils -y
+```
+
+You need to restart your master node in-order to make APPARMOR available as a kernel config security.
```
Comment (daemon1024) [inline getting-started/FAQ.md]: Restarting of node is not needed with apparmor utils. On the restart of CRI is needed.

File: getting-started/deployment_guide.md [hunk 28]
```diff
@@ -74,6 +74,8 @@ sh: 1: apt: Permission denied
 command terminated with exit code 126
 ```
 
+If you don't see Permission denied please refer [here](deployment_guide.md#debug-kubearmor-installation-issue-in-dockerized-kubernetes-environment) to debug this issue
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]:
```suggestion
If you don't see Permission denied please refer [here](deployment_guide.md#debug-kubearmor-installation) to debug this issue
```

Ig the title has changed, might need to update the link accordingly

**Review by daemon1024 — CHANGES_REQUESTED** [review]

File: .gitbook/assets/kubearmor_install_issue.png
Comment (VedRatan) [reply .gitbook/assets/kubearmor_install_issue.png:1]: done

File: getting-started/deployment_guide.md [hunk 29]
```diff
@@ -284,4 +289,84 @@ kubectl exec -it $POD -- bash -c "chroot"
...
+
+To check the status and configuration of KubeArmor, you can use the following command:
+
+```
+karmor probe
+```
+<!-- 
+<img src="../.gitbook/assets/kubearmor_install_issue.png" width="768" class="center" alt="Karmor logs"> -->
+
+```
```
Comment (daemon1024) [inline getting-started/deployment_guide.md]: Image Removed, let's remove this

**Review by daemon1024 — CHANGES_REQUESTED** [review]

**Comment by VedRatan** [comment]
@daemon1024 any reviews here?

**Review by daemon1024 — APPROVED** [review]

**Pull request merged**